openedx/features/course_experience/utils.py

```diff
@@ -216,9 +111,11 @@ def recurse_mark_auth_denial(block):
             'weight',
             'special_exam_info',
             'show_gated_sections',
-            'format'
+            'format',
+            'completion',
+            'complete',
+            'resume_block',
```

Comment:
These changes are good, but it might be nice to add a comment on the `'completion'` line to say how adding this into the requested_fields is what creates the side effect of including the BlockCompletionTransformer which is what then actually sets these three fields. I think that detail could be missed in the future and something happens where `'completion'` is removed for some reason and then `complete` and `resume_block` also break.

Reply:
Right, that's a good point. I'm also wondering whether we shouldn't hide the completion behind its waffle switch (`completion.enable_completion_tracking`) on this page. Currently we can do the following:
1. Enable the completion.
1. Publish completions on some blocks. 
1. Disable the completion.

Then we will have these checkmarks visible on the Course Outline page, despite having the completion's waffle switch set to disabled.

Reply:
Ah interesting, yes, perhaps. Although we did not gate on that before so I would say we should try and keep parity now. 

I wonder if it also might be worth a talk internally at edX about if we still need that waffle flag or if we can just move to always assuming it is active (but that's not a call for me to make, only commenting).

Reply:
Sure, thank you for clarifying :+1: 
I'll wait for the second review round then and add it (if it's desired) :)